File: typescript/rna-transcription/rna-transcription.ts
export function toRna(txt: string): string {
  let result = '';
  for (const char of txt) {
    if (RNA[char as keyof typeof RNA]) {
      result += RNA[char as keyof typeof RNA];
    } else {
      throw new Error('Invalid input DNA.');
    }
  }
  return result;
}
enum RNA {
  G = 'C',
  C = 'G',
  T = 'A',
  A = 'U'
}

File: zig/rna-transcription/rna_transcription.zig
const std = @import("std");
const mem = std.mem;

pub fn toRna(allocator: mem.Allocator, dna: []const u8) mem.Allocator.Error![]const u8 {
    var array = std.ArrayList(u8).init(allocator);
    defer array.deinit();
    for (dna) |c| {
        switch (c) {
            'G' => try array.append('C'),
            'C' => try array.append('G'),
            'T' => try array.append('A'),
            'A' => try array.append('U'),
            else => {},
        }
    }
    return array.toOwnedSlice();
}
